### DISCUSSION TITLE
Thank you

### DISCUSSION THREAD

**Discussion by mat-l** [Nailik/rhasspy_mobile#12] [General]
Great to see a new project rising to develop an Rhasspy mobile application.
I am really happy with my Rhasspy-Homeassistant setup and your app could be the missing tile in the whole installation.

Looking forward to test your application - sadly I cannot help in the implemenation due to lack of the language.

All the best!

**Comment by Nailik** [comment]
Hi 
thanks for the nice words.
 
I also use Rhasspy on Homeassistant. I tested https://github.com/razzo04/rhasspy-mobile-app but it didn't support local wake word detection and  it should run in background all the time.
First i wanted to fork it but the code is very unstructured so i started with a new project and found out that Android foreground services in Flutter are almost impossible due to the methodchannel and isolate structure.

Therefore i now switched to Kotlin Multiplatform. The Android UI is pretty much finished and Local Wake Word service is also running well. 
However MQTT and HTTP service will need quite a while because i have to read all the documentation of rhasspy to mimic the service as good as possible.

I converted this to a discussion because it's probably more useful here.

**Comment by Nailik** [comment]
MQTT and HTTP service are now integrated for the main part. Not all features are supported.

**Comment by mat-l** [comment]
I have downloaded the latest version, but I am not yet able to set it up and running with my Homeassistant and Rhasspy. But it can be also due to some local problems with my set up. ✌
From the optical point of view, the app looks already really beautiful.

**Reply by Nailik** [reply]
Please check the new release if it fixes your issues.

**Comment by Nailik** [comment]
In the Settings Tab in the app under Log Settings you can enable "Show Log" then you can read the log. This might help you to find your issues.

**Comment by Joeviocoe** [comment]
@Nailik thank you for picking up this project.  
It does appear that the original Rhasspy Mobile App project by razzo04 has been abandoned. That app does work however, but the features I really need won't be included I guess.  I tried to compile it with Flutter, but got stuck in dependency hell.  When I finally compiled an apk, it would not run.

I can help test your app, but I'm new to Android Studio development and Kotlin.  
For your 0.2 release: I am able to get it working somewhat with my HA based Rhasspy instance using MQTT.  But it hangs on STT, and TTS seems to cut off the audio too soon (I don't get the full TTS audio).
I compiled the current source code in Android Studio.  This app seems more stable (does not hang), but MQTT fails to connect.

`org.rhasspy.mobile.android W/NetworkModuleService: null: URI=tcp://org.rhasspy.mobile.observer.MutableObservable@e331e21:org.rhasspy.mobile.observer.MutableObservable@d781246
    java.lang.NoSuchFieldException: No field userInfo in class Ljava/net/URI; (declaration of 'java.net.URI' appears in /apex/com.android.art/javalib/core-oj.jar)
...`

`org.rhasspy.mobile.android E/MqttService: client not connected after attempt`
I checked the packets, and it seems the app does not even attempt to connect to my MQTT broker.  It is possible I did not build correctly.

**Reply by Nailik** [reply]
Hi @Joeviocoe thanks for testing the app.

After 0.2 i rewrote the hole main logic because in the beginning i was not sure how the rhasspy services work togther and it was a bit a mess.
Currently 0.3 waits mainly for Unit and UI Tests to deliver a more stable app because till now i had to test all the different settings by hand and this is not very efficient, later every feature should be directly include tests.

I'll check the MQTT implementation and see if i can find the issue.

**Reply by Nailik** [reply]
I actually figured out the issue, it's pretty obvious from your log.
It uses the data object instead of the value for host and port number for mqtt.
I pushed a hotfix, it will require some time until it is merged into dev (the piplines take some time)

**Reply by Joeviocoe** [reply]
I was able to test the hotfix from the PR (#99) pretty quick this morning (US time).  MQTT works now, thanks.

Right now, I'm testing the resiliency of the MQTT client connection.  It will eventually timeout after a few minutes and does not seem to reconnect on its own.  Might need to run connectClient() function somewhere in the onDisconnect() function.  

I am also testing some stability issues such as, stops responding if after speaking I wait more than a second to end the listening session manually, or let the silence detection end it.  Restarting the app is needed to get it back.

**Reply by Nailik** [reply]
What MQTT services are you using? 
I found that MQTT ist somewhat weird with rhasspy because there are a lot of possibilities where the settings end in a loop (that's why i personally would rely on the https services).
I will check your issues, so that you can use the app better and the next week(s) a lot of unit tests will be added.
Could you upload your settings file?
You can export it within the app - but make sure to remove any critical data like passwords.

**Reply by Joeviocoe** [reply]
[org.rhasspy.mobile.android_preferences.xml.txt](https://github.com/Nailik/rhasspy_mobile/files/8994786/org.rhasspy.mobile.android_preferences.xml.txt)
I'm using Remote Hermes MQTT settings for STT, Intent Recognition, and TTS.  Audio Playing and Dialog Mgmt are set to Local. and Intent Handling is "with Recognition".

I can help test and try hotfixes.  The logcat is pretty informative and the kotlin code comments are pretty good.  I should be able to troubleshoot a bit.  For example, the MQTT disconnection/timeout seems to happen only on battery power, even though the app's battery optimization is disabled.  

The "looping" does sound right.
[log.txt](https://github.com/Nailik/rhasspy_mobile/files/8994837/log.txt)

**Reply by Joeviocoe** [reply]
Switching the STT, IR, TTS services from MQTT to HTTP/API... It is indeed stable (does not hang, no loops) and works well.  

But Audio Playing does not work on the local device (response audio plays on the Rhasspy server's speaker only).  It's the same if Audio Playing is set to "Local" or "Remote...".

**Reply by Nailik** [reply]
you have to check that the response audio is set to play on the correct rhasspy id, else the app will not see the mqtt topic.
Should be posted on hermes/audioServer/<siteId>/playBytes/<requestId> (request id is irrelevant but necessary) , so you should set the topic to hermes/audioServer/mobile/playBytes/<requestId>

**Reply by Joeviocoe** [reply]
Okay so Local Play is still dependent on MQTT topic subscription?  Even if all other Rhasspy services are switched to HTTP/API?
In that case, the loop problem prevents me from using MQTT for that scenario. 
Can audio bytes (.wav) be retrieved via HTTP/API to be played on the local device?

**Reply by Nailik** [reply]
The audio playing option requires that the data is somehow transmitted to the app, there are multiple options:

directly:
- MQTT broker (even if all the other services use http/api), post the bytearray on the hermes/audioServer/mobile/playBytes/ topic
- HTTP request to the app (webserver needs to be enabled and you need to use the correct port, default is 12101, then you can post a byte array to /api/play-wav)

indirectly via http, by setting audio playing to local, tts to remote http (mqtt should work too but currently maybe not) and calling hermes/tts/say
- i have just recognized that this is missing at the moment ...

**Reply by Joeviocoe** [reply]
I can trigger this on the mobile device via the API using cURL:
`curl -X POST "http://<ServerIP>:12101/api/text-to-speech?siteId=mobile&play=true" -H  "accept: audio/wav" -H  "Content-Type: text/plain" -d "Hello World"`

This could be the way to do this without any MQTT, and without having the mobile device start its web server.

I've tried setting the TTS service in the app to, `http://<ServerIP>:12101/api/text-to-speech?siteId=mobile&play=true` but it seems to ignore the Remote HTTP all together.

**Reply by Nailik** [reply]
the text-to-speech endpoint is not implemented because it would be ambigious.
This endpoint is designed to translate the text to audio but the app is not capable of doing that, it would use the configured endpoint.

I will add a "api/say" endpoint to simplify the process of speaking audio with the app remotely (so no mqtt is required and no pre processing of text to audio) 
see https://github.com/Nailik/rhasspy_mobile/issues/103

**Reply by Joeviocoe** [reply]
I fetched the merged PR, but unsure how to configure the app to use it?  Do I need to make any changes to the Rhasspy server?  Or does this just need configuration in the app settings?  Something like `Configuration > Webserver > Enable HTTP Api > Port 12101`, then `Configuration > Text to Speech > Remote HTTP > URL: http://<RemoteServerIP>:12101/api/text-to-speech`, then `Configuration > Audio Playing > Remote HTTP > URL: http://localhost:12101/api/say` ?
Or do I keep Audio Playing as "Local"?

I've tried different combinations, but the audio response from Rhasspy still only plays on its own speaker, not the mobile device.
In `Home > Text to Speak` does output audio first on the server's speaker, and then also on the mobile device.

**Reply by Nailik** [reply]
Audio Playing should stay local.
The idea is that you call yourphoneip:120101/api/say with the text as payload.

Then the app will call the text-to-speech api to get the audio data.

The the app will call the audio configuz to play the audio (that's why it should be local).


How do you generate the audio response on your server? How do you want to play it? Maybe we think in different directions.

**Reply by Joeviocoe** [reply]
Sorry for the confusion.  This isn't the use case I was thinking about.  I was looking for the following use case (which works using MQTT normally):

1. User on mobile, clicks the microphone button
2. Speaks a command
3. Speech-to-Text using HTTP/API sends the WAV bytes to the Rhasspy server
4. Rhasspy processes WAV bytes Speech, converts to Text, sends to Intent Recognizer.
5. Intent recognizer is Home Assistant which processes the intent, sending back the Response
6. Response Text is sent back to the mobile device (somehow???)
7. Mobile device sends the Response Text back to the Text-to-Speech service on the Rhasspy server using an HTTP/API POST request
8. Rhasspy processes the Response Text and responds to the HTTP POST with WAV bytes speech.
9. Mobile device uses Audio Playing "Local" to say the Response Text on the mobile device.

I guess MQTT is a bit different using Site IDs.  The mobile device listens for responses by subscribing to the MQTT topic according to its SiteID.

I tested this new api/say service running on the mobile device.
In the Rhasspy Server web UI, I set it's Audio Playing to Remote HTTP and set the URL to `http://<mobiledeviceIP>:12101/api/say`.
I did a packet capture, and it does attempt to send WAV bytes to the phone.  

> POST /api/say HTTP/1.1
Host: x.x.x.x:12101
User-Agent: curl/7.64.0
Accept: */*
Content-Length: 39980
Expect: 100-continue
RIFF$...WAVEfmt ........"V..D.......data

But it returns `HTTP/1.1 500 Internal Server Error`

**Reply by Nailik** [reply]
I see what your problem is,
i found the issue with the `internal server error`, it's a simple fix. 

The biggest issue is that rhasspy will not send the original `siteId` inside the intent object.

You could either use
- the apps intent handling set to home assistant (the json contains the intentName, slots, text, raw text and it's own site id) and then when sending the response to someone to speak it you make a switch according to the `siteId`
- you can check if the hot word is `appsiteid_manual`  - doesn't work, rhasspy does not set the correct hot word id
- i gona fix your issue with the `internal server error` as well

i will also fix, that the app doesn't trigger the server to talk when using text-to-speech - this can't be fixed, when using the text-to-speech endpoint it will always say the text.

Best solution seems to be, to use home assistant intent handling in the app, because it sets the correct _site_id.

**Reply by Joeviocoe** [reply]
Switched to HA intent handling in the app:
`org.rhasspy.mobile.android V/HttpService: sending intent as Intent to Home Assistant
    endpoint:
    https://<serverIP>:8123/api/intent/handle
    intent:
    {"entities":[],"intent":{"confidence":1.0,"name":"GetTime"},"raw_text":"what time is it","raw_tokens":["what","time","is","it"],"recognize_seconds":0.02289750799536705,"slots":{},"speech_confidence":1.0,"text":"what time is it","tokens":["what","time","is","it"],"wakeword_id":null}`

`.ClientRequestException: Client request(POST https://<serverIP>:8123/api/intent/handle) invalid: 400 Bad Request. Text: "{"message": "Message format incorrect: extra keys not allowed @ data['entities']"}"`

Looks like the intent data formatting isn't quite right yet.

**Reply by Nailik** [reply]
Yeah i just checked in and there is a small issue that i will fix later today.
I also found out another small issue with `play-wav` endpoint, because when the data doesn't have the correct audio data size set, it will break the implementation because the app thinks it's still playing audio data.

**Reply by Joeviocoe** [reply]
`2022-06-28 12:11:45.783 24756-7093/org.rhasspy.mobile.android V/StateMachine: playAudio
2022-06-28 12:12:14.992 24756-7099/org.rhasspy.mobile.android V/StateMachine: stopListening id: mobile-default-8df8fe69-1a04-4128-a901-47d31be394fb fromMQTT: true
2022-06-28 12:12:14.993 24756-7099/org.rhasspy.mobile.android E/StateMachine: stopListening call with invalid state PlayingAudio
2022-06-28 12:12:14.994 24756-7099/org.rhasspy.mobile.android V/StateMachine: sessionEnded
2022-06-28 12:12:14.994 24756-7099/org.rhasspy.mobile.android E/StateMachine: intentTranscriptionError call with invalid state PlayingAudio
2022-06-28 12:12:14.996 24756-7099/org.rhasspy.mobile.android V/StateMachine: intentTranscribed 
2022-06-28 12:12:14.996 24756-7099/org.rhasspy.mobile.android E/StateMachine: intentTranscribed call with invalid state PlayingAudio`

One issue with the MQTT configuration leading to hangs, seems to be related to the StateMachine and playAudio not properly releasing after 30 sec timeout.  Although it should probably reset much sooner if not actively listening or receiving bytes to play.

**Reply by Nailik** [reply]
Could you test the branch
https://github.com/Nailik/rhasspy_mobile/tree/feature/105-minor-bugfixes
if you still have issues with sending an intent to home assistant?

i added some more changes, that when an intent without a name is returned, it's handled as if there was no intent recognized.

**Reply by Joeviocoe** [reply]
Thanks.  I can indeed see the response
`2022-06-28 16:07:14.051 4260-15282/org.rhasspy.mobile.android V/HttpService: sending intent received:
    {"speech": {"plain": {"speech": "The time is now 4 oh 7 P M", "extra_data": null}}, "card": {}}`
But with this configuration I still don't get audio on the mobile device, and I've tried all 3 settings for Audio Playing (Local, Remote HTTP, and Remote Hermes MQTT).

I retried setting Rhasspy base server Audio out to Remote HTTP pointing to the mobile device's web server on 12101 using the api/say endpoint.  Same 500 Internal Server Error.

Using MQTT for all services (if I don't stop the listening right when I stop talking) it still gets stuck with playAudio, and becomes unresponsive to try again.

**Reply by Joeviocoe** [reply]
the mobile app's web server api/say endpoint seems like a good use case for using it as just a satellite speaker.  not sure on how common that use case is, versus the more common using the mobile app as a full satellite rhasspy personal assistant (speak commands, execute intent, have it respond with TTS voice).  Once the MQTT hanging and timeouts are fixed, it should be ready to release as on par with razzo04's original app.

**Reply by Nailik** [reply]
I should add some information to the intent, so it knows that the origin is the app.
Then you can check this field and when this field is set you call the endpoint of the app for tts.

About mqtt: i think it would be the simplest to change the implementation so the mqtt incoming commands **only** work when dialogue management is set to mqtt.


I retried setting Rhasspy base server Audio out to Remote HTTP pointing to the mobile device's web server on 12101 using the api/say endpoint. Same 500 Internal Server Error.
- api/say is to send text
- you should use the api/play-wav endpoint, that's designed to receive a byte array

**Reply by Joeviocoe** [reply]
So the api/say endpoint on the app's webserver receives text and then converts it to speech on the local mobile device?
The api/play-wav endpoint does work if I send it wav bytes array.  Tested successful if I set the Rhasspy base Audio Play to Remote HTTP to send to the mobile device's api/play-wav endpoint. 

> add some information to the intent, so it knows that the origin is the app

Yeah, the Rhasspy base can only choose one output for Audio Playing (mine is set for aplay to a wired speaker), so the intent will need to know how to play the audio response on the origin device only.  Right now, only MQTT seems to function this way (because siteIDs are part of the topic).  If Remote HTTP could work in the same way (only the origin device plays the response) then there might not need to be a reason to use MQTT at all.

> simplest to change the implementation so the mqtt incoming commands only work when dialogue management is set to mqtt

Would this fix the instability issue of it hanging on the StateMachine condition of playAudio?  Or is that a separate issue?

**Reply by Nailik** [reply]
> So the api/say endpoint on the app's webserver receives text and then converts it to speech on the local mobile device?

It uses the defined tts endpoint and then plays the audio. The issue with this is, that the base will then play the result also, there is currently no way to migiate this issue

> Yeah, the Rhasspy base can only choose one output for Audio Playing (mine is set for aplay to a wired speaker), so the intent will need to know how to play the audio response on the origin device only. 

You should use it in whatever you use to define what happens with an intent.
For example if you use nodered, you should use nodered then to call the api /say from the app instead of the tts of your rhasspy base.

> Would this fix the instability issue of it hanging on the StateMachine condition of playAudio? Or is that a separate issue?

I have to dig further into this and test some solutions.

**Reply by Nailik** [reply]
I pushed some mqtt improvements that you can test.
But there is no fix yet, when mqtt disconnects or times out

**Reply by Joeviocoe** [reply]
tested the latest on 105 branch.  Still seems to work as before.  
For the mqtt timeout, I know you're working on implementing the keepalive and retry intervals in MqttConnectionOptions.kt
That should fix it.
Have you been able to reproduce the hanging on playAudio bug if you wait an extra 1-2 seconds after speaking to press the stop Listening button?

**Reply by Nailik** [reply]
Jeah i changed some things so when you are still "recordingintent" but mqtt already recognized an intent it will be directly used and recording etc will be stopped.

**Reply by Nailik** [reply]
hey @Joeviocoe i'm preparing v0.3 and want to know what your painpoints are at the moment besides
- mqtt loosing connection
- battery saving issue
- mqtt hanging (no timeout while waiting for answer)

When you have some points please let me know so i can fix or improve it.

**Reply by Joeviocoe** [reply]
Those would the points for v0.3  Getting the MQTT stable for use.

**Reply by Nailik** [reply]
HI @Joeviocoe i was away from this project for some time and just now added some more changes.
There are now settings for the keepalive interval as well as an reconnect interval.
I also added the possibility to set the app optimiziation from within the app so that it doesn't get killed in the background.

**Reply by Joeviocoe** [reply]
I had a chance to test it.  It seems very stable now.
One bug noticed, it hangs if using WakeWord Indicator: Visual

I think this is now good enough to by my daily driver and replace the legacy app.
The only thing I would like to see that is currently missing, is a 1x1 widget to immediately start listening for command in the background.  The Microphone overlay works great, but I'll only use that for kiosk tablets.
That, and an Android intent receiver to start listening for a command (so the Home Assistant app can send a command notification to trigger it).
Thank you.  The app is looking awesome and practically ready for release.

**Reply by Joeviocoe** [reply]
Spoke a bit too soon.
It's again hanging.   Errors say _AudioPlayer playData already playing data_, and _StateMachine intentTranscribed call with invalid state PlayingAudio_

MQTT seems to be stable and remains connected.  I can see it quickly reconnect sometimes.  So this may be a different issue from mqtt.  The Verbose logs talk a lot about the HotwordService, even though wakeword is disabled.  Don't know if that is right.

I can still reproduce by just waiting an extra 1-2 seconds after speaking before I hit the stop record button.
When it hangs, the only way to fix is to Force Stop the app.

**Reply by Nailik** [reply]
Could you attach your settings file so i can check the error?
Please also make sure to remove private data from the file

**Reply by Joeviocoe** [reply]
[org.rhasspy.mobile.android_preferences.zip](https://github.com/Nailik/rhasspy_mobile/files/9601897/org.rhasspy.mobile.android_preferences.zip)

**Reply by Nailik** [reply]
Thanks i can reproduce your issues.
Sorry for the many updates and so many problems, it's very hard to test all possibilities.
And also thanks for your patience.

**Reply by Joeviocoe** [reply]
No worries, happy to help through the dev process.

**Reply by Nailik** [reply]
I just created a new release where i hopefully fixed the mqtt issues - pipline is currently still running, should be up in a few minutes.
You should set DialogueManagement to MQTT when using MQTT for all services, because only the the session state will correctly be used.
There is now also a new option in settings under "Problem handling", this enables to force quit sessions with the microphone button when the session is running. Also the Button is greyed out when a session is currently running.

**Reply by Joeviocoe** [reply]
Unfortunately, since updating to 0.3.2-12 last night and testing today... I have not been able to get any functionality working when Dialog Management is set to MQTT (it does give 3 successive "wake" sounds when I press the mic button) . It only works on Local. Even with 0.3.2-17, only Local works.

I tried the Problem Handling Force Cancel of the session. Still, it hangs like before. Although the Mic button seems to reset and allow a new session, there is no audible wake indicator and it doesn't respond. I still have to Force Stop the entire app to reset.

**Reply by Nailik** [reply]
To further evaluate your issues i need your rhasspy base configuration, could you provide this please?

**Reply by Joeviocoe** [reply]
[rhasspy_base.txt](https://github.com/Nailik/rhasspy_mobile/files/9608751/rhasspy_base.txt)

**Reply by Nailik** [reply]
i just created a new release with lots of changes, i hope your issues will be resolved now. 
But you should uninstall the old version as the new one has breaking changes.

**Reply by Joeviocoe** [reply]
![Screenshot_20230112-154253](https://user-images.githubusercontent.com/16479297/212197221-b8dada23-3671-4f9e-8fa7-483a6a0e18b8.png)

Thank you. I've installed version 0.4.0-23 
And the new look is great. I appreciate the work you've done.

The MQTT does seem stable now.  
The attached screenshot is the expected configuration that matches how the legacy app works.  I don't think I need the web server in order for it to work since the legacy app works fine without it.

With this configuration, it does not work.

I've turned on analytics, but let me know if you need me to attach logs.

**Reply by Nailik** [reply]
yes i would need the logs attached because analytics only uploads anything if the app crashes unexpectedly.
What exactly doesn't work for you?

**Reply by Joeviocoe** [reply]
> yes i would need the logs attached because analytics only uploads anything if the app crashes unexpectedly. What exactly doesn't work for you?

```
`,{"time":"2023-01-12T23:19:54.554","severity":"Verbose","tag":"MqttService","message":"mqtt message published","throwable":null}
,{"time":"2023-01-12T23:19:54.568","severity":"Debug","tag":"DialogManagerService","message":"onAction StopListening from Local","throwable":null}
,{"time":"2023-01-12T23:19:54.570","severity":"Debug","tag":"IndicationService","message":"onRecognizingIntent","throwable":null}
,{"time":"2023-01-12T23:19:54.570","severity":"Debug","tag":"SpeechToTextService","message":"endSpeechToText sessionId: 8a955b72-6278-4254-83b2-67cd1ca93114 fromMqtt false","throwable":null}
,{"time":"2023-01-12T23:19:54.571","severity":"Verbose","tag":"MqttClient","message":"publish hermes/asr/stopListening MqttMessage(payload=[123, 34, 115, 101, 115, 115, 105, 111, 110, 73, 100, 34, 58, 34, 56, 97, 57, 53, 53, 98, 55, 50, 45, 54, 50, 55, 56, 45, 52, 50, 53, 52, 45, 56, 51, 98, 50, 45, 54, 55, 99, 100, 49, 99, 97, 57, 51, 49, 49, 52, 34, 125], msgId=2, qos=AT_MOST_ONCE, retained=false)","throwable":null}
,{"time":"2023-01-12T23:19:54.571","severity":"Debug","tag":"RecordingService","message":"stopRecording","throwable":null}
,{"time":"2023-01-12T23:19:54.571","severity":"Verbose","tag":"AudioRecorder","message":"stopRecording","throwable":null}
,{"time":"2023-01-12T23:19:54.573","severity":"Verbose","tag":"MqttService","message":"mqtt message published","throwable":null}
,{"time":"2023-01-12T23:20:04.580","severity":"Debug","tag":"DialogManagerService","message":"onAction AsrError from Local","throwable":null}
,{"time":"2023-01-12T23:20:04.581","severity":"Verbose","tag":"AudioPlayer","message":"playSoundFileResource","throwable":null}
,{"time":"2023-01-12T23:20:04.579","severity":"Debug","tag":"DialogManagerService","message":"textAsrTimeout","throwable":null}
,{"time":"2023-01-12T23:20:04.580","severity":"Debug","tag":"LocalAudioService","message":"playErrorSound","throwable":null}
,{"time":"2023-01-12T23:20:04.580","severity":"Debug","tag":"IndicationService","message":"onError","throwable":null}
,{"time":"2023-01-12T23:20:04.639","severity":"Verbose","tag":"MqttClient","message":"publish hermes/error/asr MqttMessage(payload=[123, 34, 115, 101, 115, 115, 105, 111, 110, 73, 100, 34, 58, 34, 56, 97, 57, 53, 53, 98, 55, 50, 45, 54, 50, 55, 56, 45, 52, 50, 53, 52, 45, 56, 51, 98, 50, 45, 54, 55, 99, 100, 49, 99, 97, 57, 51, 49, 49, 52, 34, 125], msgId=2, qos=AT_MOST_ONCE, retained=false)","throwable":null}
,{"time":"2023-01-12T23:20:04.640","severity":"Verbose","tag":"MqttService","message":"mqtt message published","throwable":null}
,{"time":"2023-01-12T23:20:04.641","severity":"Debug","tag":"DialogManagerService","message":"onAction SessionEnded from Local","throwable":null}
,{"time":"2023-01-12T23:20:04.641","severity":"Debug","tag":"IndicationService","message":"onIdle","throwable":null}
,{"time":"2023-01-12T23:20:04.641","severity":"Verbose","tag":"MqttClient","message":"publish hermes/dialogueManager/sessionEnded MqttMessage(payload=[123, 34, 115, 105, 116, 101, 73, 100, 34, 58, 34, 109, 111, 98, 105, 108, 101, 34, 44, 34, 115, 101, 115, 115, 105, 111, 110, 73, 100, 34, 58, 34, 110, 117, 108, 108, 34, 125], msgId=2, qos=AT_MOST_ONCE, retained=false)","throwable":null}
,{"time":"2023-01-12T23:20:04.642","severity":"Verbose","tag":"MqttService","message":"mqtt message published","throwable":null}
,{"time":"2023-01-12T23:20:13.701","severity":"Debug","tag":"IndicationService","message":"onWakeWordDetected","throwable":null}
,{"time":"2023-01-12T23:20:13.701","severity":"Debug","tag":"DialogManagerService","message":"onAction WakeWordDetected from Local wakeWord: manual","throwable":null}
,{"time":"2023-01-12T23:20:13.702","severity":"Debug","tag":"LocalAudioService","message":"playWakeSound","throwable":null}
,{"time":"2023-01-12T23:20:14.367","severity":"Debug","tag":"DialogManagerService","message":"onAction StartSession from Local","throwable":null}
,{"time":"2023-01-12T23:20:14.368","severity":"Debug","tag":"WakeWordService","message":"stopDetection","throwable":null}
,{"time":"2023-01-12T23:20:14.368","severity":"Debug","tag":"DialogManagerService","message":"onAction SessionStarted from Local","throwable":null}
,{"time":"2023-01-12T23:20:14.373","severity":"Debug","tag":"DialogManagerService","message":"onAction StartListening from Local sendAudioCaptured: false","throwable":null}
,{"time":"2023-01-12T23:20:14.374","severity":"Debug","tag":"WakeWordService","message":"stopDetection","throwable":null}
,{"time":"2023-01-12T23:20:14.374","severity":"Debug","tag":"IndicationService","message":"onListening","throwable":null}
,{"time":"2023-01-12T23:20:14.374","severity":"Debug","tag":"SpeechToTextService","message":"startSpeechToText sessionId: 45b72941-3e7c-4eb4-9356-d377f1ded33f fromMqtt false","throwable":null}
,{"time":"2023-01-12T23:20:14.376","severity":"Debug","tag":"RecordingService","message":"startRecording","throwable":null}
,{"time":"2023-01-12T23:20:15.955","severity":"Debug","tag":"DialogManagerService","message":"onAction StopListening from Local","throwable":null}
,{"time":"2023-01-12T23:20:15.956","severity":"Debug","tag":"IndicationService","message":"onRecognizingIntent","throwable":null}
,{"time":"2023-01-12T23:20:15.956","severity":"Debug","tag":"SpeechToTextService","message":"endSpeechToText sessionId: 45b72941-3e7c-4eb4-9356-d377f1ded33f fromMqtt false","throwable":null}
,{"time":"2023-01-12T23:20:15.956","severity":"Debug","tag":"RecordingService","message":"stopRecording","throwable":null}
,{"time":"2023-01-12T23:20:25.960","severity":"Debug","tag":"IndicationService","message":"onError","throwable":null}
,{"time":"2023-01-12T23:20:25.959","severity":"Debug","tag":"DialogManagerService","message":"textAsrTimeout","throwable":null}
,{"time":"2023-01-12T23:20:25.960","severity":"Debug","tag":"DialogManagerService","message":"onAction AsrError from Local","throwable":null}
,{"time":"2023-01-12T23:20:25.961","severity":"Debug","tag":"LocalAudioService","message":"playErrorSound","throwable":null}
,{"time":"2023-01-12T23:20:26.055","severity":"Debug","tag":"DialogManagerService","message":"onAction SessionEnded from Local","throwable":null}
,{"time":"2023-01-12T23:20:26.055","severity":"Debug","tag":"IndicationService","message":"onIdle","throwable":null}` #

**Reply by Joeviocoe** [reply]
Manually triggering to start listening, speaking the command, pressing again to End, takes a few seconds, then error tone.

**Reply by Joeviocoe** [reply]
```
[D] TIME: 2023-01-12T16:30:41.707789 Starting mqtt... TAG: MQTT
[D] TIME: 2023-01-12T16:30:41.710037 Connecting to mqtt... TAG: MQTT
[D] TIME: 2023-01-12T16:30:41.742607 connected TAG: MQTT
[I] TIME: 2023-01-12T16:30:41.743215 Mosquitto client connected TAG: MQTT
[I] TIME: 2023-01-12T16:30:42.673744 Start recording TAG: APP
[D] TIME: 2023-01-12T16:30:42.675512 Send port: SendPort TAG: MQTT
[D] TIME: 2023-01-12T16:30:43.104422 generate new data session TAG: MQTT
[I] TIME: 2023-01-12T16:30:45.340033 Stop recording TAG: APP
[D] TIME: 2023-01-12T16:30:45.469085 received topic: hermes/asr/textCaptured TAG: MQTT
[D] TIME: 2023-01-12T16:30:45.542529 received topic: hermes/nlu/intentParsed TAG: MQTT
[I] TIME: 2023-01-12T16:30:45.543465 Recognized intent: GetTime TAG: DIALOGUE
[D] TIME: 2023-01-12T16:30:45.543771 Waiting for intent to be handle TAG: MQTT
[D] TIME: 2023-01-12T16:30:46.855144 received topic: hermes/audioServer/mobile/playBytes/4f05b473-d50c-4175-b5ff-8ed225eebbad TAG: MQTT
[D] TIME: 2023-01-12T16:30:46.856186 received audio TAG: MQTT
[D] TIME: 2023-01-12T16:30:46.872048 Received audio to play TAG: DIALOGUE
[D] TIME: 2023-01-12T16:30:47.357690 StopRecording request TAG: DIALOGUE
[I] TIME: 2023-01-12T16:30:47.360029 Stop recording TAG: APP
```

This is the log from the legacy app, that is working.

**Reply by Nailik** [reply]
@Joeviocoe i think i was able to reproduce your issue and fixed it in 0.4.1 .

**Reply by Joeviocoe** [reply]
I was able to update through the Play store and the latest version seems your best yet.
Seems to be working stable so far.
Occasionally, it'll get stuck and takes a minute before it will work again.

Still testing. But so far, so good.

**Reply by Nailik** [reply]
Would be interesting to know when/why it gets stuck.

**Reply by Nailik** [reply]
@Joeviocoe i just released a pretty big update, should fix a lot of stuck issues.
